src/marks/line.js

```diff
@@ -26,25 +26,49 @@ const defaults = {
 export class Line extends Mark {
   constructor(data, options = {}) {
     const {x, y, z, curve, tension} = options;
+
+    // When the line mark is used with a "sphere" curve and a projection
+    // (supposedly spherical), keep x and y as numbers.
+    const sphere = curve === "sphere";
     super(
       data,
       {
-        x: {value: x, scale: "x"},
-        y: {value: y, scale: "y"},
+        x: {value: x, scale: sphere ? undefined : "x"},
+        y: {value: y, scale: sphere ? undefined : "y"},
         z: {value: maybeZ(options), optional: true}
       },
       options,
       defaults
     );
     this.z = z;
-    this.curve = Curve(curve, tension);
+    if (sphere) this.sphere = true;
+    else this.curve = Curve(curve, tension);
     markers(this, options);
   }
   filter(index) {
     return index;
   }
   render(index, scales, channels, dimensions, context) {
     const {x: X, y: Y} = channels;
+
+    let shape;
+    if (this.sphere) {
+      const {projection} = context;
+      if (projection === undefined) throw new Error("the sphere curve requires a projection");
+      const path = geoPath(projection);
+      shape = (I) =>
+        path({
+          type: "LineString",
+          coordinates: Array.from(I, (i) => [+X[i], +Y[i]]).filter(([x, y]) => !isNaN(x + y))
+        });
```

Comment:
We could do this faster by invoking the projection stream directly (calling _stream_.lineStart, _stream_.point, _stream_.lineEnd) rather than going through geoPath.

Also, this will coerce nulls to zero rather than skipping them. We might want to use coerceNumbers here, but we should at least be consistent.

Reply:
groupIndex already does filtering for undefined values, so we can check if `i` is -1 here instead. I’ll push some commits.